src/framework/smoke_job.hpp

```diff
@@ -154,9 +154,10 @@ class SmokeJob {
     return task;
   }
 
-  static std::list<SmokeJob> createJobsFromJson(const SmokeFlags flags, bool&
+  static std::list<std::shared_ptr<SmokeJob>> createJobsFromJson(
```

Comment:
Anyway no memory leaks will happen since we don't have here circular pointers.. Any other risk?